apiBase/APIRequestUtils.es6.js

```diff
@@ -145,38 +149,60 @@ const normalizeRequest = (res, req) => {
   return req;
 };
 
-const handleRequestIfFailed = (apiOptions, err, res, req, reject) => {
-  if (!(err || (res && !res.ok))) { return; }
-
-  getEmitter(apiOptions).emit(Events.error, err, req);
-  const errorHandler = apiOptions.defaultErrorHandler || reject;
-  errorHandler(err || 500);
-  return true;
-};
-
-const handle = (apiOptions, resolve, reject, err, res, req, query, method,
+const handle = (apiOptions, resolve, reject, err, res, req, method, path, query,
     parseBody, parseMeta) => {
 
   req = normalizeRequest(res, req);
 
-  if (handleRequestIfFailed(apiOptions, err, res, req, reject)) {
+  if (handleRequestIfFailed(apiOptions, err, res, req, method, path, reject)) {
     return;
   }
 
   getEmitter(apiOptions).emit(Events.response, req, res);
 
-  const apiResponse = tryParseResponse(reject, res, req, method, query, parseBody, parseMeta);
+  const apiResponse = tryParseResponse(reject, res, req, method, path, query, parseBody, parseMeta);
 
   getEmitter(apiOptions).emit(Events.result, apiResponse);
   resolve(apiResponse);
 };
 
-const tryParseResponse = (reject, res, req, method, query, parseBody, parseMeta) => {
+const handleRequestIfFailed = (apiOptions, err, res, req, method, path, reject) => {
+  if (!(err || (res && !res.ok))) { return; }
+
+  // Error handling ported from 1X, this could be revisited in more depth
+  // but it seems like getting ResponseErrors to be thrown is more important
+  // for now.
+  if (err) {
+    getEmitter(apiOptions).emit(Events.error, err, req);
+
+    if (err && err.timeout) {
+      err.status = 504;
+    }
+
+    // This is a good example of error-handling we should reconsider soon.
+    // The api sending back a 302 for a subreddit might be a good indicator
+    // that we should really be issuing a 302 to a url that has the canonical
+    // capitalization of a subreddit for SEO or any number of other things.
+    // 1X's comment:
+    // Sometimes the API decides to send back a 302 instead of a 404; for
+    // example, unfound subs will redirect you to the search page.
+    if (method === 'get' && status === 302) {
```

Comment:
This makes me nervous. Is there ever a case where the 302 is real and that we should follow it to some resource? I guess that'd be captured by the error checking above?